src/mavlink-router/endpoint.cpp

```diff
@@ -81,7 +81,7 @@ int Endpoint::handle_read()
 
     while ((r = read_msg(&buf, &target_sysid, &target_compid, &src_sysid,
                          &src_compid, &msg_id)) > 0) {
-        if (allowed_by_filter(msg_id) && allowed_by_dropout())
+        if (allowed_by_filter(msg_id) && allowed_by_dropout() && allowed_by_dedup(&buf))
```

Comment:
You have moved the filters (in previous commits) after the point where the list of "known components" is updated. This means, that a message, which is rejected by any incoming filter, will update that list. But IMHO this should not be the case. "Rejected" messages shouldn't update the list of known source/component-IDs.
(Then, de-duplication will solve most of the issues we have with out multiple modems/ antennas on one device.)

Reply:
Hmm yes, that was unintentional. Better for filtered messages not to alter state, that's part of the reason to filter them after all.